### ISSUE TITLE
✨ Update to work experience - UI changes

### ISSUE THREAD

**Issue by gobyrne** [GCTC-NTGC/gc-digital-talent#11864]
## ✨ Feature

We want to see a history of an employee's work in government for talent management. To this end, we want to collect group and level on the work experience. We also want to be able to match seniority in other orgs to those in gov, so we collect seniority when we do not collect group and level. Afterwards we will eventually deprecate collecting group and level in the "government employee" section.

## 🕵️ Details

There are a lot of things that show and hide, and display in various situations. Here's a list of all the changes and small interactions (but please also c[lick through the design](https://www.figma.com/proto/piRkRtWAHheoEX5Jq8d9u7/Create%2FEdit-an-experience-(Applicants)?node-id=487-58844&node-type=canvas&t=7ZVdFKtLK183qYms-0&scaling=min-zoom&content-scaling=fixed&page-id=472%3A70951)) :

### All experience
1. New question for all work experience "Employment category: External, Gov, CAF"
2. Start date and end date are moved to the bottom of form
3. Add a sentence to the description for "Highlight additional details" section stays at the bottom
4. In the designs the copy for "Current experience" is changed but honestly, we were just too lazy to revert all the designs back to match production. You do not need to change the labels for that checkbox.

### External
5. Add "size of org" question
6. Add "Seniority of the role" question (with "Self-employed" as an option)
7. **Read only state** - only needs to add those two fields, otherwise unchanged

### Gov
8. Instead of org, we ask for "Department, agency or other government organization" and it should be a Drop-down of the departments we have in the database.
9. Ask "Employment type"
10.  For Casual, term, or indeterminate we ask "Group" and "Level"
11. For indeterminate and term, we ask "Position type"
11.5 (**Gray added this late**) For term, we do **not** give the option for "substantive" in "Position type"
12. For Student, Casual, Term, Indeterminate-Acting, Indeterminate-Assignment, Indeterminate-Secondment, we ask for "Expected end date" if "Are you currently in this role?" is selected. After consultation with @petertgiles it was decided to store this data in the same field, just swap the label.
13. For Contractor we ask "Seniority of the role" (but no "Self-employed" option)
14. Contractor we ask "Contractor type" and if "firm or agency" is selected, we ask "Contracting firm or agency"
15. **Read only state** - The "Department, agency or other government organization" is shown where Organization normally shows up
16. **Read only state** - Add "Government of Canada" between Work experience and Dates
17. **Read only state** - Except if they are a Contractor. In that case replace "Government of Canada" with "Contractor"
18. **Read only state** - in the case of an end-date that's in the future. The text (expected end date) should be added
19. **Read only state** - Show employment type for all
20. **Read only state** - Show employment type, Position type and classification when collected

### CAF
21. Ask "Employment type" (Regular or Reserve)
22. Ask "Military force"
23. Ask Rank category
24. **Read only state** - The "Military force" ais shown where Organization normally shows up
25. **Read only state** - Add "Canadian Armed Forces" between Work experience and Dates
24. **Read only state** - Add "Employment type" and "Rank category" fields

## 🎨 Design
https://www.figma.com/design/piRkRtWAHheoEX5Jq8d9u7/Create%2FEdit-an-experience-(Applicants)?node-id=472-70951&t=NbjSJSxQjUSo26FF-1

## ✅ Acceptance Criteria

- [ ] Can view, add, and edit work experiences in career timeline
- [ ] Can view, add, and edit work experiences in application process
- [ ] Can view work experiences in admin profile view of career timeline
- [ ] Can view work experiences in admin view of snapshot - simple and new experience types
- [ ] Update backend validation to force entering new fields in the API
- [ ] Design review (Jerbo or Gray)

Out of scope: Classification picker with locking icon.  Use existing picker like on "create account" flow.
Out of scope: Downloadable files
Out of scope: Changing organization field to a searchable dropdown. It can stay as a text input.

## 🛑 Blockers

```[tasklist]
### Blocked By
- [ ] https://github.com/GCTC-NTGC/gc-digital-talent/issues/11868
```
